docs/how-to-guides.md

```diff
@@ -32,6 +32,199 @@ Everything about test data generation and its adjustable parameters can
+   OwnMobileNetModel model of Tensorflow and do not implement our own model definition.
+   ```yaml
+   defaults:
+     - op_train_base.yaml@_here_
+     - /shared/datasets@data_train: dataset_kaggle_flowers.yaml
+     - /shared/datasets@data_validation: dataset_kaggle_flowers.yaml
+     - _self_
+   train_params:
+     epochs: 5
+   data_description:
+     _target_: niceml.data.datadescriptions.clsdatadescription.ClsDataDescription
+     classes:
+     - "daisy"
+     - "roses"
+     use_binary: true
+     target_size:
+       _target_: niceml.utilities.imagesize.ImageSize
+       width: 180
+       height: 180
+   
+   data_train:
+     datainfo_listing:
+       sub_dir: train
+     set_name: train
+     shuffle: true
+   data_validation:
+     datainfo_listing:
+       sub_dir: validation
+     set_name: validation
+   model:
+     _target_: niceml.dlframeworks.tensorflow.models.mobilenet.OwnMobileNetModel
+     final_activation: sigmoid
+   model_load_custom_objects:
+     _target_: niceml.mlcomponents.modelcompiler.modelcustomloadobjects.ModelCustomLoadObjects
+   learner:
+     _target_: niceml.dlframeworks.tensorflow.learners.defaultlearner.DefaultLearner
+     model_compiler:
+       _target_: niceml.dlframeworks.tensorflow.modelcompiler.defaultmodelcompiler.DefaultModelCompiler
+       loss: binary_crossentropy
+       # Optimizer used in the experiment
+       optimizer:
+         _target_: tensorflow.keras.optimizers.Adam
+         learning_rate: 0.0001
+       metrics: [ "accuracy" ]   
+   ```
+
+   4. You might have noticed, that we do not give a plain directory definition to tell
+   niceml where the training images are stored. Instead, we define the datasets within
+   a configuration yaml in `configs/shared/datasets/dataset_kaggle_flowers.yaml`. For now,
+   we will use as the same configuration as in `dataset_cls_test.yaml`
+6. Configure prediction operation
+   1. We create `configs/ops/prediction/op_prediction_cls_binary_flowers.yaml` and use
+   `op_prediction_cls.yaml` as a template.
+   2. Here, we only have to change the definition of the training, validation and test datasets with
+   the yaml we wrote in the step before (`dataset_kaggle_flowers.yaml`). 
+   The prediction configuration should look like this:
+   
+```yaml
+      defaults:
+   - /shared/datasets@datasets.validation: dataset_kaggle_flowers.yaml
+   - /shared/datasets@datasets.test: dataset_kaggle_flowers.yaml
+   - /shared/datasets@datasets.train_eval: dataset_kaggle_flowers.yaml
+   - prediction_handler: prediction_handler_vector.yaml
+   - datasets: datasets_generic_default.yaml
+   - op_prediction_base.yaml@_here_
+   - _self_
+```
+
+7. Start the pipeline using 
+```python
+niceml train configs/jobs/job_train/job_train_cls/job_train_cls_binary_flowers.yaml
+```
+
+## How to add a custom model
+In order to define a custom model in niceML we can make use of the niceML `ModelFactory` class
```

Comment:
Suggestion:
In order to define a custom model in niceML we can make use of the niceML `ModelFactory` class
and its `create_model` function. To create a new model, implement your by inheriting from the ModelFactory and configuring the functions you need. Here is an example: